nym-vpn/ui/src-tauri/src/commands/connection.rs

```diff
@@ -115,12 +123,27 @@ pub async fn connect(
             ),
         )
         .ok();
-    });
+    }
+
+    // Start exit message listener
+    // This will listen for the (single) exit message from the VPN client and update the UI accordingly
+    debug!("starting exit listener");
+    register_exit_listener(app, state.inner().clone(), vpn_exit_rx)
```

Comment:
I suggest something like "spawn_exit_listener". With word "register_" it sounds a little like you're registering a callback. In this case we are starting a separate worker task